packages/near-membrane-base/src/membrane.ts

```diff
@@ -0,0 +1,1467 @@
+    getPointer: PrimitiveOrPointer,
+    setPointer: PrimitiveOrPointer
+) => boolean;
+type CallableDeleteProperty = (targetPointer: Pointer, key: PropertyKey) => boolean;
+type CallableDescriptorCallback = (
+    configurable: boolean | symbol,
+    enumerable: boolean | symbol,
+    writable: boolean | symbol,
+    valuePointer: PrimitiveOrPointer,
+    getPointer: PrimitiveOrPointer,
+    setPointer: PrimitiveOrPointer
+) => void;
+type CallableGetOwnPropertyDescriptor = (
+    targetPointer: Pointer,
+    key: PropertyKey,
+    foreignCallableDescriptorCallback: CallableDescriptorCallback
+) => void;
+type CallableGetPrototypeOf = (targetPointer: Pointer) => PrimitiveOrPointer;
+type CallableHas = (targetPointer: Pointer, key: PropertyKey) => boolean;
+type CallableIsExtensible = (targetPointer: Pointer) => boolean;
+type CallableOwnKeys = (
+    targetPointer: Pointer,
+    foreignCallableKeysCallback: (...args: (string | symbol)[]) => void
+) => void;
+type CallablePreventExtensions = (targetPointer: Pointer) => boolean;
+export type CallableSetPrototypeOf = (
+    targetPointer: Pointer,
+    protoValueOrPointer: PrimitiveOrPointer
+) => boolean;
+type CallableGetTargetIntegrityTraits = (targetPointer: Pointer) => number;
+type CallableHasOwnProperty = (targetPointer: Pointer, key: PropertyKey) => boolean;
+type CallableLinkIntrinsics = (...reflectiveIntrinsicPointers: Pointer[]) => void;
+type CallableLinkUnforgeables = (...unforgeablePointers: Pointer[]) => void;
+export type CallableInstallLazyDescriptors = (
+    targetPointer: Pointer,
+    ...keyAndEnumTuple: PropertyKey[]
+) => void;
+export type CallableEvaluate = (sourceText: string) => void;
+export type GetTransferableValue = (value: any) => PrimitiveOrPointer;
+export type GetSelectedTarget = () => any;
+export type HooksCallback = (
+    getSelectedTarget: GetSelectedTarget,
+    getTransferableValue: GetTransferableValue,
+    callableEvaluate: CallableEvaluate,
+    callableInstallLazyDescriptor: CallableInstallLazyDescriptors,
+    callablePushTarget: CallablePushTarget,
+    callableApply: CallableApply,
+    callableConstruct: CallableConstruct,
+    callableDefineProperty: CallableDefineProperty,
+    callableDeleteProperty: CallableDeleteProperty,
+    callableGetOwnPropertyDescriptor: CallableGetOwnPropertyDescriptor,
+    callableGetPrototypeOf: CallableGetPrototypeOf,
+    callableHas: CallableHas,
+    callableIsExtensible: CallableIsExtensible,
+    callableOwnKeys: CallableOwnKeys,
+    callablePreventExtensions: CallablePreventExtensions,
+    callableSetPrototypeOf: CallableSetPrototypeOf,
+    callableGetTargetIntegrityTraits: CallableGetTargetIntegrityTraits,
+    callableHasOwnProperty: CallableHasOwnProperty,
+    callableLinkIntrinsics: CallableLinkIntrinsics,
+    callableLinkUnforgeables: CallableLinkUnforgeables,
+    globalThisPointer: Pointer
+) => void;
+
+export type DistortionCallback = (target: ProxyTarget) => ProxyTarget;
```

Comment:
Can the types be split into their own types file?

Reply:
We can discuss this offline